sdcm/utils/k8s.py

```diff
@@ -409,10 +409,13 @@ def patch_kubectl_auth_config(cls, config, auth_type, cmd: str, args: list):
 
     @staticmethod
```

Comment:
```
    @staticmethod
    def wait_for_pods_state(kluster,
                            total_pods: Union[int, Callable],
                            state='Running',  # Running | Ready
                            timeout: float,
                            namespace: str,
                            sleep_between_retries: int = 10):

        assert isinstance(total_pods, (int, float)) or callable(total_pods), "total_pods should be number or callable"

        @timeout_decor(message=f"Wait for {total_pods} pod(s) from {namespace} namespace to reach the {state} state...",
                       timeout=timeout * 60,
                       sleep_time=sleep_between_retries)
        def wait_cluster_is_ready():
            # To make it more informative in worst case scenario made it repeat 5 times, by readiness_timeout // 5
            result = kluster.kubectl(
                f"wait --timeout={readiness_timeout // 5}m --all --for=condition={state} pod",
                namespace=namespace,
                timeout=readiness_timeout * 60 // 5 + 10)
            count = result.stdout.count('condition met')
            if isinstance(total_pods, (int, float)):
                if total_pods != count:
                    raise RuntimeError('Not all pods reported')
            elif callable(total_pods):
                if not total_pods(count):
                    raise RuntimeError('Not all pods reported')
            else:
                raise ValueError(f"total_pods should be number or callable. Found {type(total_pods)}")

        wait_cluster_is_ready()

   @staticmethod
    def wait_for_pods_readiness(kluster, total_pods: Union[int, Callable], readiness_timeout: float, namespace: str,
                                sleep_between_retries: int = 10):
        wait_for_pods_state(... state='Ready', ...)

   @staticmethod
    def wait_for_pods_running(kluster, total_pods: Union[int, Callable], readiness_timeout: float, namespace: str,
                              sleep_between_retries: int = 10):
        wait_for_pods_state(... state='Running', ...)
```

?

Reply:
I guess I'll name it `wait_for_pods_with_condition`